### ISSUE TITLE
.hist().relabel() doesn't work

### ISSUE THREAD

**Issue by basnijholt** [ioam/holoviews#157]
doesn't work:
```
hv.Image(np.random.rand(10,10)).hist().relabel('Test')
```

does work:
```
hv.Image(np.random.rand(10,10)).relabel('Test').hist()
```

EDIT:
Just realised that with ```hv.Image(np.random.rand(10,10)).hist().relabel('Test', depth=1)``` it does work, so don't know if this is a bug or not?

**Comment by jlstevens** [comment]
I don't think this is a bug.

After you call ``.hist()`` you have a data structure containing both the ``Image`` and ``Histogram`` (this container is called an ``AdjointLayout`` and we aren't proud of it). This means you need to increase the depth in contrast to the case where you relabel and *then* call ``.hist()``.

In general, I would recommend setting the group/label in the constructor and avoid using ``relabel`` unless you actually need to *re*label something (as opposed to initializing a label).

**Comment by basnijholt** [comment]
I agree it's not a bug. 

Although it might be useful to provide some message when one tries to label a ```hist()```, such that when one encounters this issue, they'll at least have some guidance.

**Comment by philippjfr** [comment]
Hmm, AdjointLayout may be another instance of an object where we could set a different default depth since group and labels on the ``AdjointLayout`` itself simply aren't useful.

**Comment by jlstevens** [comment]
@philippjfr I agree - this should be an easy fix.

**Comment by jlstevens** [comment]
Should be fixed in 0ec373d.
